exporter/collector/logs.go

```diff
@@ -109,8 +116,51 @@ func (l *LogsExporter) Shutdown(ctx context.Context) error {
 }
 
 func (l *LogsExporter) PushLogs(ctx context.Context, ld plog.Logs) error {
+	entries, err := l.mapper.createEntries(ld)
+	if err != nil {
+		return err
+	}
+
+	errors := []error{}
+	entry := 0
+	currentBatchSize := 0
+	// Send entries in WriteRequest chunks
+	for len(entries) > 0 {
+		// TODO(damemi): Add integration test for batch request processing
+		if entry < len(entries) && currentBatchSize+proto.Size(entries[entry]) < defaultMaxRequestSize {
+			// if adding the current entry to the current batch doesn't go over the request size,
+			// increase the index and account for the new request size, then continue
+			currentBatchSize = currentBatchSize + proto.Size(entries[entry])
```

Comment:
can we combine into a single proto.Size call per-entry?

Reply:
Yep, I tried being lazy to guard for panics here by throwing the `entry < len(entries)` check first into this if block but I just broke that out